clang/test/CIR/CodeGen/atomic-thread-fence.c

```diff
@@ -0,0 +1,149 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --check-prefix=CIR --input-file=%t.cir %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm %s -o %t.ll
+// RUN: FileCheck --check-prefix=LLVM --input-file=%t.ll %s
+
+
+struct Data {
+  int value;
+  void *ptr;
+};
+
+typedef struct Data *DataPtr;
+
+void applyThreadFence() {
+  __atomic_thread_fence(__ATOMIC_SEQ_CST);
+}
+
+// CIR-LABEL: cir.func no_proto @applyThreadFence
+// CIR:   %0 = cir.const #cir.int<5> : !s32i
+// CIR:   cir.atomic.fence(sync_scope = system, ordering = seq_cst)
```

Comment:
Get rid of the parens, see https://llvm.github.io/clangir/Dialect/cir-style-guide.html
This could be simplified such that the output is a bit more simple, example:
`cir.atomic.fence system seq_cst`

Reply:
Looks like this refactoring needs to be done for other operations too, since all of them have the parenthesis.

Reply:
Yes, there's tech debt, but it's orthogonal from this PR